src/Mvc/Mvc.Core/src/Infrastructure/ObjectResultExecutor.cs

```diff
@@ -151,23 +151,19 @@ private Task ExecuteAsyncCore(ActionContext context, ObjectResult result, Type?
         private static void InferContentTypes(ActionContext context, ObjectResult result)
         {
             Debug.Assert(result.ContentTypes != null);
-            if (result.ContentTypes.Count != 0)
```

Comment:
Can we keep this check for non-ProblemDetails responses?

i.e.
```C#
if (result.value is ProblemDetails)
{
   result.ContentTypes.Add("application/problem+json");
   result.ContentTypes.Add("application/problem+xml");
}
else if (result.ContentTypes.Count == 0 && !string.IsNullOrEmpty(resposeContentType))
{
  result.ContentTypes.Add(responseContentType);
}
```

Reply:
Sure, Will change :)

Reply:
@pranavkm done 👍